crypto/evp/evp_rand.c

```diff
@@ -624,7 +636,7 @@ unsigned int EVP_RAND_get_strength(EVP_RAND_CTX *ctx)
 {
     unsigned int res;
 
-    if (!evp_rand_lock(ctx))
+    if (!evp_rand_lock(ctx, 1))
```

Comment:
Read once and cache is a better approach here, strength doesn't change.

Reply:
As mentioned in my comment below we only seem to hit this on a reseed event, so i'm not too sure its worth worrying about too much. But can do so if you insist.

Reply:
I really don't think the DRBGs are abusing this lock and I'd prefer implementing the read and cache rather than adding a new public provider API.

I don't actually object to leaving this as a write lock, it should be rarely called.